### PR TITLE
Move the ingress relation under requires

### PR DESCRIPTION
**Pull request by sed-i**
## Issue
Cannot relate grafana to traefik.

## Solution
- Move the ingress relation under requires
- Add itest

## Context
```
$ juju relate trfk graf
ERROR no relations found
```

## Testing Instructions
Relate grafana to traefik.

## Release Notes
Move the ingress relation under requires

### CODE REVIEW

**Review by simskij — APPROVED** [review]

File: src/charm.py [hunk 1]
```diff
@@ -1069,11 +1069,11 @@ def external_url(self) -> str:
...
 
     @property
-    def _ingress_config(self) -> str:
+    def _ingress_config(self) -> dict:
         """Build a raw ingress configuration for Traefik."""
         fqdn = urlparse(socket.getfqdn()).path
-        domain = fqdn.split(
-            "{}.{}.".format(self.model.unit.name.replace("/", "-"), self.model.name)
+        domain = re.split(
+            r"^{}\..*?{}\.".format(self.model.unit.name.replace("/", "-"), self.model.name), fqdn
```
Comment (Abuelodelanada) [inline src/charm.py:1076]: Reading regular expressions is not always easy.... 

What about this:

```python
pattern = r"^{}\..*?{}\.".format(self.model.unit.name.replace("/", "-"), self.model.name)
domain = re.split(pattern, fqdn)
```

**Review by Abuelodelanada — APPROVED** [review]

File: src/charm.py (hunk 1 above)
Comment (sed-i) [reply src/charm.py:1076]: Fixed in next commit.

Comment (rbarry82) [reply src/charm.py:1076]: Either works for me, though the difference here mostly seems to be in assigning it as a separate variable. If we're going to do that, we may as well `re.compile()` while we're at it

**Comment by rbarry82** [comment]
Let's merge as soon as CI passes

**Review by rbarry82 — APPROVED** [review]

**Comment by sed-i** [comment]
A seemingly unrelated itest repeatedly fails now: #137

**Pull request merged**